controllers/snapshot/snapshot_adapter.go

```diff
@@ -0,0 +1,444 @@
+	bindingComponents := a.getApplicationSnapshotComponents(components)
+
+	applicationSnapshotEnvironmentBinding := &appstudioshared.ApplicationSnapshotEnvironmentBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: bindingName + "-",
+			Namespace:    namespace,
+		},
+		Spec: appstudioshared.ApplicationSnapshotEnvironmentBindingSpec{
+			Application: applicationName,
+			Environment: environmentName,
+			Snapshot:    appSnapshot.Name,
+			Components:  *bindingComponents,
+		},
+	}
+
+	err := a.client.Create(a.context, applicationSnapshotEnvironmentBinding)
+	return applicationSnapshotEnvironmentBinding, err
+}
+
+// getAutoReleasePlansForApplication returns the ReleasePlans used by the application being processed. If matching
+// ReleasePlans are not found, an error will be returned. A ReleasePlan will only be returned if it has the
+// release.appstudio.openshift.io/auto-release label set to true or if it is missing the label entirely.
+func (a *Adapter) getAutoReleasePlansForApplication(application *hasv1alpha1.Application) (*[]releasev1alpha1.ReleasePlan, error) {
+	releasePlans := &releasev1alpha1.ReleasePlanList{}
+	labelSelector := labels.NewSelector()
+	labelRequirement, err := labels.NewRequirement("release.appstudio.openshift.io/auto-release", selection.NotIn, []string{"false"})
+	if err != nil {
+		return nil, err
+	}
+	labelSelector = labelSelector.Add(*labelRequirement)
+
+	opts := &client.ListOptions{
+		Namespace:     application.Namespace,
+		FieldSelector: fields.OneTermEqualSelector("spec.application", application.Name),
+		LabelSelector: labelSelector,
+	}
+
+	err = a.client.List(a.context, releasePlans, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return &releasePlans.Items, nil
+}
+
+// getReleasesWithApplicationSnapshot returns all Releases associated with the given applicationSnapshot
+// In the case the List operation fails, an error will be returned.
+func (a *Adapter) getReleasesWithApplicationSnapshot(applicationSnapshot *appstudioshared.ApplicationSnapshot) (*[]releasev1alpha1.Release, error) {
+	releases := &releasev1alpha1.ReleaseList{}
+	opts := []client.ListOption{
+		client.InNamespace(applicationSnapshot.Namespace),
+		client.MatchingFields{"spec.applicationSnapshot": applicationSnapshot.Name},
+	}
+
+	err := a.client.List(a.context, releases, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &releases.Items, nil
+}
+
+// createReleaseForReleasePlan creates the Release for a given ReleasePlan.
+// In case the Release can't be created, an error will be returned.
+func (a *Adapter) createReleaseForReleasePlan(releasePlan *releasev1alpha1.ReleasePlan, applicationSnapshot *appstudioshared.ApplicationSnapshot) (*releasev1alpha1.Release, error) {
```

Comment:
It would be a great idea to extract this to a `release/releaseplan.go` file.

Reply:
Ack, extracted the function to the `releaseplan.go` under the `release` package.